File: web/src/lib/project-display.ts
import type { ProjectSummary } from "@/lib/types";

/** Имя в сайдбаре — не topic ноды «Тема ролика». */
export function projectDisplayName(
  p: { title?: string | null; topic?: string | null; slug: string },
): string {
  const title = (p.title ?? "").trim();
  if (title) return title;
  const topic = (p.topic ?? "").trim();
  if (topic) return topic;
  return p.slug;
}

File: app/services/project_display.py
"""Отображаемое имя проекта в UI (сайдбар, тосты) — не topic для пайплайна."""

from __future__ import annotations

from app.models import Project


def project_display_name(project: Project) -> str:
    """title → topic (legacy) → slug."""
    title = (getattr(project, "title", None) or "").strip()
    if title:
        return title
    topic = (project.topic or "").strip()
    if topic:
        return topic
    return project.slug
